cli/tests/test_git_remotes.rs

```diff
@@ -13,10 +13,18 @@
 // limitations under the License.
 
 use std::fs;
+use std::path::Path;
 use std::path::PathBuf;
 
 use crate::common::TestEnvironment;
 
+fn read_git_remote_config(repo_path: &Path) -> String {
+    let git_config = fs::read_to_string(repo_path.join(".jj/repo/store/git/config"))
+        .or_else(|_| fs::read_to_string(repo_path.join(".git/config")))
+        .unwrap();
+    git_config[git_config.find("[remote ").unwrap()..].to_owned()
```

Comment:
nit: I think it's better to mask out platform-dependent values explicitly instead of skipping until `[remote `. It'll ensure that non-remote tables are retained.

BTW, thanks for splitting the first two patches. It's pleasant to review this PR.

Reply:
Done.

And thank you! I realized it’d be good to see the diff from the `git2` → `gix` switch in the commits :)